code-challenges.js

```diff
@@ -10,6 +10,61 @@ var people = [
+  let eachObject = arrayOfObjects[index]
+  let nameOfPerson = eachObject.name
+  //split the person's name which is a string into 2 strings, the first and last name. It will be split at the space
+  let firstAndLastNames = nameOfPerson.split(" ")
+  //take in the array that is created when those strings are split, map over it, and uppercase the first letter for each item in the array. set it equal to a variable
+  let uppercasedNames = firstAndLastNames.map(value => { 
+      return value.charAt(0).toUpperCase() + value.slice(1) })
+  // join the items of these arrays of the first and last names into single strings and return this for the mapped function
+  let arrayOfNames = uppercasedNames.join(" ")
+//arrayOfNames returns an array with each capitalized name. this next line adds a key to the original object with the capitalized name.
+//eachObject is now an array of objects 
+    eachObject.capitalizedName = arrayOfNames
+    return arrayOfObjects
+  })
+  let finalSentences = arrayOfObjects.map(objectInArray => {
+    return `${objectInArray.capitalizedName} is a ${objectInArray.occupation}.`})
+  // return `${eachObject.capitalizedName} is a ${eachObject.occupation}.`
+  // console.log(arrayOfObjects.occupation)
+  // return `${} is a ${arrayOfObjects.occupation}.`
+  return finalSentences
+}
+
+console.log(sentenceReturner(people))
+
+
+//refactor to not use the index potentially. 
+
+sentenceReturner2 = (arrayOfObjects) => {
+  //map over the array of objects and take in the items of the array and their indices
+  arrayOfObjects.map((arrayItem) => {
+    //set a variable for the key in the array of objects that we need to access (name)
+  let nameOfPerson = arrayItem.name
+  //split the person's name which is a string into 2 strings, the first and last name. It will be split at the space and return an array with the first and last name at index 0 & 1 respectively.
+  let firstAndLastNames = nameOfPerson.split(" ")
+  //take in the array that is created when those strings are split, map over it, and uppercase the first letter for each item in the array. set it equal to a variable
+  let uppercasedNames = firstAndLastNames.map(value => { 
+      return value.charAt(0).toUpperCase() + value.slice(1) })
+  // join the items of these arrays of the first and last names into single strings and return this for the mapped function
+  let arrayOfNames = uppercasedNames.join(" ")
+//arrayOfNames returns an array with each capitalized name. this next line adds a key to the original object called capitalized name with the arrayOfNames as its values.
+    arrayItem.capitalizedName = arrayOfNames
+    //return the updated arrayOfObjects that includes the capitalized names.
+    return arrayOfObjects
+  })
+  //this will map over the arrayOfObjects to access each object in the array. then we can use the keys & values within those objects to return the final sentences.
+  let finalSentences = arrayOfObjects.map(objectInArray => {
```

Comment:
```suggestion
  return arrayOfObjects.map(objectInArray => {
```
You can just return this array without declaring a variable since you're not reusing the variable for anything else